### PR TITLE
[CardTitle] Additional functionality

### PR DESCRIPTION
**Pull request by kylegach**
### Description

- Add support for meta information
- Add demo meta info demo example
- Refactor examples to use a shared DemoLayout
- Update snapshots

### Motivation and context

The style guide specifies that CardTitle needs to optionally display a category title. Also, though the style guide specifies a version of the Card/CardTitle that has an avatar, we have opted to not yet build that functionality, as the design is very much in flux.

### Screenshots or videos, if appropriate

screen shot 2017-07-24 at 11 58 30 am

### How has this been tested?

Confirmed to render correctly in all supported browsers.

### Types of changes

- New feature (non-breaking change which adds functionality)

### Checklist

* [x] Renders and functions properly in [all supported browsers](https://github.com/mineral-ui/mineral-ui#browser-support)
* [x] Automated tests written and passing
* [x] [Accessibility](http://webaim.org/intro) and [inclusivity](https://24ways.org/2016/what-the-heck-is-inclusive-design/) considered
* [x] Rendering performance (initial load time & 60fps) and [perceived performance](http://blog.teamtreehouse.com/perceived-performance) considered
* [x] Documentation created or updated

### How does this PR make you feel?

Crank

### CODE REVIEW

File: packages/card/src/__demo__/examples/card-title/withCategoryTitle.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+});
+
+export default {
+  title: 'With a category title',
+  backgroundColor: mineralTheme.color_gray_10,
+  description:
+    'If you provide a category title, it will display above the title, which will automatically display in the minor style.',
+  scope: { Card, CardBlock, CardTitle, DemoLayout },
+  source: `<DemoLayout>
+    <Card>
```
Comment (brentertz) [inline packages/card/src/__demo__/examples/card-title/withCategoryTitle.js]: @coldpour Did you ever get to experiment at all with `dedent` here?

File: packages/card/src/CardTitle.js [hunk 2]
```diff
@@ -24,7 +24,9 @@ import CardRow from './CardRow';
 import cardTheme from './cardTheme';
 
 type Props = {|
-  /** Rendered content of the component */
+  /** Category title displayed above the title */
+  categoryTitle?: string,
```
Comment (brentertz) [inline packages/card/src/CardTitle.js]: I agree with @coldpour on this one.  `categoryTitle` seems odd and unintuitive.

File: packages/card/src/__demo__/examples/card-title/withCategoryTitle.js (hunk 1 above)
Comment (coldpour) [inline packages/card/src/__demo__/examples/card-title/withCategoryTitle.js]: i never did, but given the number of times i've caught indentation issues, maybe i should... do we want a library to do our indentation for us at runtime or do we want to have it correct in the code too?

i could also put in a console warning if the source has a newlines or tabs in the wrong places

File: packages/card/src/CardTitle.js (hunk 2 above)
Comment (kylegach) [inline packages/card/src/CardTitle.js]: @brentertz, @coldpour — Curious for your thoughts on this alternative approach: https://github.com/mineral-ui/mineral-ui/pull/187/commits/a4d60be4a97b92aadbd2a39c0005e3838f782c93

Comment (kylegach) [inline packages/card/src/CardTitle.js]: Nevermind. I simply renamed `categoryTitle` to `meta`, instead.

**Pull request merged**